poly-commitment/src/commitment.rs

```diff
@@ -500,6 +500,34 @@ where
 }
 
 impl<G: CommitmentCurve> SRS<G> {
+    /// Commits a polynomial, and adds the blinding factor to it.
+    /// It is not a blinding commitment, as the blinding factor is scaled by 1 instead of a random value.
+    /// This is useful for when the polynomial can be zero,
+    /// in which case the commitment would be the point at infinity if no blinding factor was added.
+    /// (And points at infinity are annoying to handle in circuits.)
+    pub fn commit_public(&self, poly: &DensePolynomial<G::ScalarField>) -> BlindedCommitment<G> {
```

Comment:
It seems like you've reinvented a apecialised `mask_custom`. Can we use that instead?